src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/ReflectionXmlFormatReader.cs

```diff
@@ -465,9 +576,91 @@ private bool ReflectionTryReadPrimitiveArray(Type type, Type itemType, int array
             return true;
         }
 
-        internal void ReflectionReadGetOnlyCollection(XmlReaderDelegator xmlReader, XmlObjectSerializerReadContext context, XmlDictionaryString itemName, XmlDictionaryString itemNamespace, CollectionDataContract collectionContract)
+        internal void ReflectionReadGetOnlyCollection(XmlReaderDelegator xmlReader, XmlObjectSerializerReadContext context, XmlDictionaryString itemName, XmlDictionaryString itemNs, CollectionDataContract collectionContract)
         {
-            throw new NotImplementedException();
+            ReflectionInitArgs(xmlReader, context, null, null);
+            _arg2CollectionItemName = itemName;
+            _arg3CollectionItemNamespace = itemNs;
+
+            if (_arg0XmlReader.IsStartElement(_arg2CollectionItemName, _arg3CollectionItemNamespace))
+            {
+                object resultCollection = _arg1Context.GetCollectionMember();
+                if (resultCollection == null)
+                {
+                    XmlObjectSerializerReadContext.ThrowNullValueReturnedForGetOnlyCollectionException(collectionContract.UnderlyingType);
+                }
+
+                Type collectionType = collectionContract.UnderlyingType;
+                bool isArray = (collectionContract.Kind == CollectionKind.Array);
+                int arraySize = 0;
+                if (isArray)
+                {
+                    arraySize = ((Array)resultCollection).Length;
+                }
+
+                string itemNameStr = collectionContract.ItemName;
+                string itemNsStr = collectionContract.StableName.Namespace;
+
+                for (int index = 0; index < int.MaxValue; index++)
```

Comment:
I'm afraid we don't have a good quota for this. Here we are reading collection items from XML. We don't know the size of the collection at this point.